core/src/autogluon/core/scheduler/seq_scheduler.py

```diff
@@ -0,0 +1,273 @@
+        self.searcher: BaseSearcher = self.get_scheduler_(searcher, train_fn, **kwargs)
+        self.init_limits_(kwargs)
+        self.metadata = {
+            'search_space': train_fn.kwspaces,
+            'search_strategy': self.searcher,
+            'stop_criterion': {
+                'time_limits': self.time_out,
+                'max_reward': self.max_reward},
+            'resources_per_trial': self.resource
+        }
+
+        args = kwargs.get('args')
+
+    def init_limits_(self, kwargs):
+        if kwargs.get('num_trials') is None:
+            assert kwargs.get('time_out') is not None, "Need stopping criterion: Either num_trials or time_out"
+        self.num_trials = kwargs.get('num_trials', 9999)
+        self.time_out = kwargs.get('time_out')
+        if self.num_trials is None:
+            assert self.time_out is not None, \
+                "Need stopping criterion: Either num_trials or time_out"
+
+    def get_scheduler_(self, searcher, train_fn, **kwargs) -> BaseSearcher:
+        if searcher is 'local_sequential_auto':
+            searcher = 'auto'
+        scheduler_opts = {}
+        if searcher is 'auto':
+            searcher = 'bayesopt'
+            scheduler_opts = {'scheduler': 'fifo'}
+        search_options = kwargs.get('search_options')
+        if isinstance(searcher, str):
+            if search_options is None:
+                search_options = dict()
+            _search_options = search_options.copy()
+            _search_options['configspace'] = train_fn.cs
+            _search_options['reward_attribute'] = kwargs['reward_attr']
+            _search_options['resource_attribute'] = kwargs.get('time_attr', None)
+            # Adjoin scheduler info to search_options, if not already done by
+            # subclass
+            if 'scheduler' not in _search_options:
+                _search_options['scheduler'] = 'fifo'
+            searcher = searcher_factory(searcher, **{**scheduler_opts, **_search_options})
+        else:
+            assert isinstance(searcher, BaseSearcher)
+        return searcher
+
+    def run(self, **kwargs):
+        """Run multiple trials given specific time and trial numbers limits.
+        """
+        self.searcher.configure_scheduler(self)
+
+        self.training_history = OrderedDict()
+        self.config_history = OrderedDict()
+
+        time_start = time.time()
+
+        avg_trial_run_time = None
+
+        r = range(self.num_trials)
+        for i in (tqdm(r) if self.num_trials < 1000 else r):
+            trial_start_time = time.time()
+            self.run_trial(task_id=i)
+            trial_end_time = time.time()
+
+            if self.max_reward and self.get_best_reward() >= self.max_reward:
+                logger.log(20, f'\tMax reward is reached')
+                break
+
+            if self.time_out is not None:
+                avg_trial_run_time = self.get_average_trial_time_(i, avg_trial_run_time, trial_start_time, trial_end_time)
```

Comment:
It would be good to keep track of failed runs somehow, since we don't want them contributing to average trial time estimates, and it will be useful for debugging.

Reply:
Also, can you document this in the class header? This is a non-obvious behaviour.

Reply:
Re-worked this logic; fixed a defect related to handling failed runs.